pkg/querier/queryrange/queryrangebase/definitions/interface.go

```diff
@@ -32,28 +32,16 @@ type Merger interface {
 
 // Request represents a query range request that can be process by middlewares.
 type Request interface {
-	// GetStart returns the start timestamp of the request in milliseconds.
-	GetStart() time.Time
-	// GetEnd returns the end timestamp of the request in milliseconds.
-	GetEnd() time.Time
-	// GetStep returns the step of the request in milliseconds.
-	GetStep() int64
-	// GetQuery returns the query of the request.
-	GetQuery() string
-	// GetCachingOptions returns the caching options.
-	GetCachingOptions() CachingOptions
-	// WithStartEnd clone the current request with different start and end timestamp.
-	WithStartEnd(start time.Time, end time.Time) Request
+	resultscache.Request
```

Comment:
This refactoring bothers me a little for two reasons. 
1. conceptually, this is confusing. What has `resultscache.Request` to do anything with main `Request` interface type used in all the read path for all the queries?

Not sure about coupling these two types this way. It should be the other way around, "results cache using main request interface".

2. Embedding types are comparatively make it hard for readability IMHO. Given this is one of the main interface types.

the way I think about this is, if circular dependencies are causing problems to reuse these types, then better we move these types completely to different package.

Makes more sense for both `queryrange` and `resultscache` package uses exactly same type say `query.Request` instead of coupling them with two different types + embedding. Seems complex to me.

wdyt @salvacorts ?

Reply:
(we discussed this in Slack).

I ended up separating the definition of `queryrangebase.Request/Response` from `resultscache.Request/Response`. The return type of the `WithStartEnd` method was being problematic (ambiguous and duplicated definition) So I ended up following the [same pattern as LokiRequest][1]: have `WithStartEnd` to implement `queryrangebase.Request` and `WithStartEndTime` to implement `resultscache.Request`.

[1]: https://github.com/grafana/loki/blob/5d84adfe372997dc09bc9f2425b1cff1998bfbbb/pkg/querier/queryrange/codec.go#L59-L71